dashboard/test/models/data_doc_test.rb

```diff
@@ -0,0 +1,48 @@
+require 'test_helper'
+
+class DataDocTest < ActiveSupport::TestCase
+  test "can create data doc" do
+    data_doc = create :data_doc
+    assert data_doc.name
+    assert data_doc.key
+  end
+
+  class KeyConstraintTests < ActiveSupport::TestCase
```

Comment:
I stole all of these from `programming_expression_test.rb` ... is there a better way?

Reply:
since these tests are covering functionality in `validate_key_format` defined in `CurriculumHelper`,  ideally tests of this detail would be in a test file for that specific helper. if you want to deduplicate this, you could try following these instructions to write a test specifically for the helper: https://guides.rubyonrails.org/testing.html#testing-helpers the helper test seems to implicitly include the module into the test class, but if that doesn't seem to work you could explicitly do something like https://github.com/code-dot-org/code-dot-org/blob/4dcd93c0e5a7c0c002e4ff617ddfc58dc439b4ba/dashboard/test/helpers/pd/application/pd_workshop_helper_test.rb#L4-L5 to get access to `validate_key_format` so that you can test it.

for this to work, you will probably need to stub out the `key` method as though it is a method on your test class.